Mathlib/LinearAlgebra/RootSystem/Defs.lean

```diff
@@ -130,3 +130,47 @@ def reflection : M ≃ₗ[R] M :=
 /-- The reflection associated to a coroot. -/
 def coreflection : N ≃ₗ[R] N :=
   Module.reflection (P.root_coroot_two i)
+
+section pairs
+
+variable (i j : ι)
+
+/-- This is the pairing between roots and coroots. -/
+def pairing : R := P.toLin (P.root i) (P.coroot j)
+
+/-- The Coxeter Weight of a pair gives the weight of an edge in a Coxeter diagram, when it is
+finite.  It is `4 cos² θ`, where `θ` describes the dihedral angle between hyperplanes. -/
+def coxeterWeight : R := pairing P i j * pairing P j i
+
+/-- Two roots are orthogonal when they are fixed by each others' reflections. -/
+def IsOrthogonal : Prop := pairing P i j = 0 ∧ pairing P j i = 0
+
+/-- A pair of roots is nonsymmetrizable if the existence of a Weyl-invariant form on their span is
+obstructed.  When `R` has no zero divisors, this happens when exactly one root is fixed by the
+other's reflection. -/
+def IsNonSymmetrizable : Prop := coxeterWeight P i j = 0 ∧ ¬ IsOrthogonal P i j
+
+/-- Two roots are parallel if their Coxeter weight is exactly 4.  A linearly independent pair of
+parallel roots generates an infinite dihedral group of reflections on their span, and their span
+has an invariant singular line. -/
+def IsParallel : Prop := coxeterWeight P i j = 4
+
+variable [LinearOrderedCommRing R]
+
+/-- An imaginary pair is one whose Coxeter weight is negative.  Any symmetrization yields an
+invariant form where one of the roots has negative norm and the other has positive norm. Root
+systems with imaginary pairs are not often considered, since the geometry of reflection is
+cumbersome. -/
+def IsImaginary : Prop := coxeterWeight P i j < 0
+
+/-- A pair of roots is definite if there is a unique (up to scalars) definite reflection-invariant
+form on their span. This happens precisely when the Coxeter weight is strictly between `0` and `4`.
+-/
+def IsDefinite : Prop := 0 < coxeterWeight P i j ∧ coxeterWeight P i j < 4
+
+/-- A pair of roots is ultraparallel if its Coxeter weight is strictly greater than 4.
+Symmetrization induces a nondegenerate indefinite form on the span, and reflections generate an
+infinite dihedral group. -/
+def IsUltraParallel : Prop := 4 < coxeterWeight P i j
```

Comment:
That's a lot of definitions without API.

Reply:
Should I cut some of it out for a later PR?

Reply:
@ScottCarnahan maybe you already have more material? In that case, would you mind opening another PR that has all the material? That helps with reviewing.

For the current PR, I would suggest keeping the first 3 defs `pairing`, `coxeterWeight`, and `IsOrthogonal`, and writing a few API lemmas about them like `coxeterWeight_swap` and `IsOrthogonal_comm` and `IsOrthogonal.symm`.

I guess the other definitions will come with more serious API.